src/lib/Scenes/City/Components/AllEvents.tsx

```diff
@@ -28,11 +29,38 @@ export class AllEvents extends React.Component<Props, State> {
   }
 
   componentDidMount() {
-    this.updateSections()
+    this.updateSections(this.props)
   }
 
-  updateSections = () => {
-    const { buckets, cityName } = this.props
+  componentWillReceiveProps(nextProps) {
+    if (
+      !isEqual(
+        this.props.buckets.saved.map(g => g.node.is_followed),
+        nextProps.buckets.saved.map(g => g.is_followed)
+      ) ||
+      !isEqual(
+        this.props.buckets.closing.map(g => g.node.is_followed),
+        nextProps.buckets.closing.map(g => g.is_followed)
+      ) ||
+      !isEqual(
+        this.props.buckets.museums.map(g => g.node.is_followed),
+        nextProps.buckets.museums.map(g => g.is_followed)
+      ) ||
+      !isEqual(
+        this.props.buckets.opening.map(g => g.node.is_followed),
+        nextProps.buckets.opening.map(g => g.is_followed)
+      ) ||
+      !isEqual(
+        this.props.buckets.galleries.map(g => g.node.is_followed),
+        nextProps.buckets.galleries.map(g => g.is_followed)
+      )
```

Comment:
Seeing as these all look for the same prop, you could make this a bit DRYer by creating a function that takes the before and after props and the bucket key to compare.

Reply:
Yup you could do something like (double check bool logic for me)

```javascript
const shouldUpdate = ["saved", "closing", "museums", "opening", "closing"].map(key => {
  return !isEqual(
        this.props.buckets[key].map(g => g.node.is_followed),
        nextProps.buckets[key].map(g => g.is_followed)
      )
  )
}).some(a => a)

if (shouldUpdate) { this.updateSections(nextProps)) }
```

Reply:
ah nice, ill update now.

Reply:
Actually even better
`Object.keys(this.props.buckets).map`

Reply:
Problem with that is it would include the `fairs`, we don't want all the bucket keys